### PR TITLE
feat: add interlinear PDF export pipeline

### LINKED ISSUE #109: pdf interlinear export
Along with the weekly data export, we can export a PDF formatted interlinear

### PR DESCRIPTION
**Pull request by delgado-jacob**
closes: https://github.com/globalbibletools/platform/issues/109

## Feature 

  For Translators:
  - Export interlinear PDFs directly from the Language Settings page
  - Choose specific books and chapter ranges, or export everything
  - Select between "Standard" (word-by-word) and "Parallel" (two-column) layouts
  - Download links with automatic expiration tracking

  For Snapshots:
  - Automatic PDF generation when language snapshots are created
  - PDFs are archived alongside snapshot data for historical record

 ## Architecture

  New export Module (src/modules/export/)
  - InterlinearPdfGenerator - Core PDF rendering using PDFKit with SBL Hebrew/Greek fonts
  - ExportRequestRepository - Tracks export requests, status, and download URLs
  - ExportStorageRepository - S3/LocalStack integration for PDF storage
  - Server actions for requesting exports and polling status
  - React components for the export UI with real-time status updates

  Background Jobs
  - export_interlinear_pdf - Generates PDFs asynchronously, merges multi-book exports
  - cleanup_exports - Purges expired exports from storage
  - create_snapshot_interlinear_pdf - Generates archival PDFs during snapshot creation

  Database Changes
  - New export_request table tracking request lifecycle (pending → in progress → complete/failed)
  - New export_request_book junction table for multi-book exports
  - Three new job types registered

 ## Key Design Decisions

  1. Async Generation: PDFs are generated in background jobs rather than blocking the UI, with polling for status updates
  2. Per-Book Chunking with Merge: Multi-book exports generate individual PDFs per book, then merge them using pdf-lib—this prevents memory issues with large exports
  3. Script Detection: Automatically detects Hebrew vs Greek source text to apply correct fonts and RTL/LTR layout
  4. Expiring Downloads: Export URLs expire (configurable, default 7 days) with automatic cleanup job
  5. Snapshot Integration: Snapshot creation now triggers interlinear PDF generation, archived at snapshots/{languageId}/{snapshotId}/interlinear.pdf

## Testing

  - Unit tests for PDF generator, jobs, actions, and React components
  - Integration tests for repository layer
  - LocalStack integration tests for S3 operations
  - Test coverage for error paths and edge cases

### CODE REVIEW

File: src/modules/snapshots/jobs/createSnapshotJob.ts [hunk 1]
```diff
@@ -41,10 +43,15 @@ export async function createSnapshotJob(job: CreateSnapshotJob) {
...
-    environment: process.env.NODE_ENV === "production" ? "prod" : "local",
+    environment: getStorageEnvironment(),
     snapshot,
   });
   await snapshotRepository.create(snapshot);
+  await enqueueJob(SNAPSHOT_JOB_TYPES.CREATE_SNAPSHOT_INTERLINEAR_PDF, {
+    languageId: snapshot.languageId,
+    languageCode: language.code,
+    snapshotId: snapshot.id,
+  });
```
Comment (arrocke) [inline src/modules/snapshots/jobs/createSnapshotJob.ts]: The snapshot system is meant for backup/restore purposes, not to export for general consumption. We can remove PDF generation from the snapshot module

File: db/migrations/25-12-06-add-export-request.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
+create type export_request_status as enum ('PENDING', 'IN_PROGRESS', 'COMPLETE', 'FAILED');
+
+create table export_request (
```
Comment (arrocke) [inline db/migrations/25-12-06-add-export-request.sql]: I'd prefer to track all of this in the `payload` and `data` columns of the `job` table. Both of those columns can hold arbitrary JSON. Jobs are ephemeral so I find referential integrity of job data to be less important and having a single mechanism for tracking job progress is useful. Do you have any concerns with this approach?

File: src/modules/export/pdf/InterlinearPdfGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+import fs from "fs";
+
+export interface PdfGeneratorOptions {
+  pageSize?: "letter" | "a4";
+  layout: "standard" | "parallel";
+  direction?: "ltr" | "rtl";
+  header?: { title?: string; subtitle?: string };
+  footer?: { generatedAt?: Date; pageOffset?: number; pageTotal?: number };
+  sourceScript?: "hebrew" | "greek";
+}
```
Comment (arrocke) [inline src/modules/export/pdf/InterlinearPdfGenerator.ts]: For this first version, I'd like to simplify this to a single button for the user. Let's use your standard layout and export all books by default even if there are no glosses for that book yet. This is the MVP of this feature and we can expand options later.

**Review by arrocke — COMMENTED** [review]
Wow thanks for the contribution! There is a lot of good stuff in here, but a PR this size is very difficult to review. I'm going to treat this like a prototype to work through product/design decisions together. Then let's chop up the PR up into smaller PRs for easier technical review. There will be a lot to work through and this will make it less overwhelming to review and faster to get changes shipped for testing.

In the future, I encourage you to seek clarity on vague tickets before starting work and open PRs with smaller changes. We have a feature flag system to be able to do this with incomplete features.

We can use this PR to discuss the bigger picture of this feature, so I will leave some high level feedback here to start that discussion. From there, here is a way we can break this work up into smaller PRs for easier review:
1. **Introduce localstack:** I'm thrilled you are introducing this. There are a number of other places where we could take advantage of this. This PR could simply introduce localstack and its initialization scripts.
2. **Basic export job infrastructure:** Set up the UI to trigger the new job and report it's status. At this stage the job is a noop. This new UI can be behind a feature flag.
3. **Upload logic:** Add the logic to upload an empty PDF in the job.
4. **PDF generation:** Add the MVP PDF generation logic to the job. This will be the meat of the work, but we'll be able to focus on just that logic now that all of the infrastructure is taken care of.

Please don't be discouraged by this feedback. My goal is to clarify how we can best work together. I'm grateful for the work you've put in thus far to develop this feature. Please let me know if you have any questions.
